components/camel-bean/src/main/java/org/apache/camel/component/bean/BeanPropertiesFunction.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.bean;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.CamelContextAware;
+import org.apache.camel.NoSuchBeanException;
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.spi.PropertiesFunction;
+import org.apache.camel.spi.Registry;
+import org.apache.camel.support.ObjectHelper;
+
+@org.apache.camel.spi.annotations.PropertiesFunction("bean")
+public class BeanPropertiesFunction implements PropertiesFunction, CamelContextAware {
+    private CamelContext camelContext;
+
+    @Override
+    public String getName() {
+        return "bean";
+    }
+
+    @Override
+    public String apply(String remainder) {
+        String[] beanNameAndMethodName = remainder.split("\\.");
+        String beanName = beanNameAndMethodName[0];
+        String methodName = beanNameAndMethodName[1];
+
+        Registry registry = getCamelContext().getRegistry();
+        Object bean = registry.lookupByName(beanName);
+        if (bean == null) {
+            throw new NoSuchBeanException(beanName);
+        }
+
+        String answer = "";
+        try {
+            answer += (String) ObjectHelper.invokeMethodSafe(methodName, bean);
```

Comment:
Use camel typ e converter to. convert to a string value, as it may invoke a method that returns a boolean, int etc.